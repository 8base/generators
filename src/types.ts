import { TableSchema } from '@8base/utils';

export interface GeneratorsCommonConfig {
  deep?: number;
  withMeta?: boolean;
}

export type GeneratorsConfig = {
  includeColumns?: string[],
} & GeneratorsCommonConfig;

export interface GeneratorsData {
  tablesList: TableSchema[];
  tableName: string;
  screenName?: string;
}

export interface ScreenTable {
  id?: string;
  tableName: string;
  screenName: string;
  routeUrl?: string;
  tableFields?: string[];
  formFields?: string[];
}
