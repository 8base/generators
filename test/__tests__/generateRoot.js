

import { generateRoot } from '../../src';


it('should generate update form by the table name', () => {
  const generatedRoot = generateRoot([{
    screenName: 'post',
    routeUrl: '/post',
  }, {
    screenName: 'author',
    routeUrl: '/author',
  }]);

  expect(generatedRoot).toMatchSnapshot();
});
