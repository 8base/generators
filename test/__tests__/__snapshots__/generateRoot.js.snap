// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate update form by the table name 1`] = `
"import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';

import { ProtectedRoute } from './shared/components';
import { MainPlate, ContentPlate, Nav } from './components';
import { Auth } from './routes/auth';
import { Post } from './routes/post';
import { Author } from './routes/author';

export const Root = () => (
  <Switch>
    <Route path=\\"/auth\\" component={Auth} />
    <ProtectedRoute>
      <MainPlate>
        <Nav.Plate color=\\"BLUE\\">
          <Nav.Item to=\\"/post\\" label=\\"Post\\" />
          <Nav.Item to=\\"/author\\" label=\\"Author\\" />
        </Nav.Plate>
        <ContentPlate>
          <Switch>
            <ProtectedRoute exact path=\\"/post\\" component={Post} />
            <ProtectedRoute exact path=\\"/author\\" component={Author} />
          </Switch>
        </ContentPlate>
      </MainPlate>
    </ProtectedRoute>
  </Switch>
);
"
`;
